theano/sandbox/gpuarray/basic_ops.py

```diff
@@ -996,6 +996,8 @@ def get_params(self, node):
     def perform(self, node, axis_and_tensors, out_, ctx):
         out, = out_
         axis = int(axis_and_tensors[0])
+        if axis < 0:
+            axis += axis_and_tensors[1].ndim
```

Comment:
I'm pretty sure that this is done inside of concatenate.  It doesn't hurt to have it here though.